pkg/template/helm_template.go

```diff
@@ -167,3 +178,50 @@ func (t *HelmTemplate) writeFile(dstPath, subPath string, content []byte) (strin
 
 	return newPath, nil
 }
+
+type helmTemplateCmdArgs struct {
+	binaryName string
+	args       []string
+}
+
+const helmBinaryName = "helm"
+
+// DEPRECATED
+const helm2BinaryName = "helmv2"
+const helm2ChartSpecVersion = "v1"
+
+// auxiliary struct used for Chart.yaml unmarshalling
+type chartSpec struct {
+	APIVersion string
+}
+
+// DEPRECATED. This method will not be required once support for Helm 2 is dropped
+// Returns the Helm Binary Name and the arguments required to be passed to the "helm template" subcommand
+// The returned values depend on the ApiVersion property inside the Chart.yaml file.
+// apiVersion==v1 will fallback to old Helm 2 binary and command format.
+func helmTemplateCmdLookup(releaseName, chartPath, namespace string) (*helmTemplateCmdArgs, error) {
+	// Load [chartPath]/Chart.yaml and inspect apiVersion value.
+	bs, err := ioutil.ReadFile(path.Join(chartPath, "Chart.yaml"))
+	if err != nil {
+		return nil, fmt.Errorf("helmTemplateCmdLookup: %w", err)
+	}
+
+	var chartSpec chartSpec
+	if err := yaml.Unmarshal(bs, &chartSpec); err != nil {
+		return nil, fmt.Errorf("helmTemplateCmdLookup: %w", err)
+	}
+
+	// By default, use Helm 3+ format except for chart.apiSpec=v1
+	res := &helmTemplateCmdArgs{
+		binaryName: helmBinaryName,
+		args:       []string{"template", releaseName, chartPath, "--namespace", namespace, "--include-crds"},
+	}
+
+	// DEPRECATED: Helm V2 will be removed in a future release
+	if chartSpec.APIVersion == helm2ChartSpecVersion {
+		res.binaryName = helm2BinaryName
+		res.args = []string{"template", chartPath, "--name", releaseName, "--namespace", namespace}
+	}
```

Comment:
Probably can do the following here instead:

```suggestion
	
	res := &helmTemplateCmdArgs{}
	// DEPRECATED: Helm V2 will be removed in a future release
	if chartSpec.APIVersion == helm2ChartSpecVersion {
		res.binaryName = helm2BinaryName
		res.args = []string{"template", chartPath, "--name", releaseName, "--namespace", namespace}
	} else {
	      // By default, use Helm 3+ format except for chart.apiSpec=v1
	      res.binaryName = helmBinaryName,
	      res.args = []string{"template", releaseName, chartPath, "--namespace", namespace, "--include-crds"}
	}
```

Reply:
Thanks @danielhelfand for the review.

I am happy to change that but I'd love to know more about the suggestion so I can tailor my code better in the future, I am also quite rusty so this is a good exercise to learn. Is it that what I did is not Golang idiomatic? Do you find it more difficult to understand? Is there a bug?

Some background context about the reason I did it like that is that I am expecting the Helm v2 to go away soon and in my code it would mean to just remove the override down below, leaving the struct initialization intact. I know this is quite nitpick and not a strong reason :P

I appreciate it! Thank you!

Reply:
i would actually go with @migmartri's version as well due to the reason mentioned

> I am expecting the Helm v2 to go away soon and in my code it would mean to just remove the override down below, leaving the struct initialization intact

though if we are being nitpicky, i would have also reinitialized res to a new struct to avoid picking up any unintended fields. not a big deal here.

Reply:
> though if we are being nitpicky, i would have also reinitialized res to a new struct to avoid picking up any unintended fields. not a big deal here.

Yup, that was my thought but understand the reasoning. Works for me.